### PR TITLE
fix managedcluster watch do not work

### PR DESCRIPTION
**Pull request by ldpliu**
## Related issue(s)
https://issues.redhat.com/browse/ACM-13443
Fixes #

## Tests
* [ ] Unit/function tests have been added and incorporated into `make unit-tests`.
* [ ] Integration tests have been added and incorporated into `make integration-test`.
* [x] E2E tests have been added and incorporated into `make e2e-test-all`.
* [x] List other manual tests you have done.
manually test the update and create managedcluster, the annotations will be added.

### CODE REVIEW

File: operator/pkg/controllers/crd/crd_controller.go [hunk 1]
```diff
@@ -269,22 +269,13 @@ func watchClusterManagementAddOnPredict() predicate.TypedPredicate[*v1alpha1.Clu
...
-			if e.ObjectNew.GetLabels()[constants.GlobalHubOwnerLabelKey] !=
-				constants.GHOperatorOwnerLabelVal {
-				return false
-			}
-			// only requeue when spec change, if the resource do not have spec field, the generation is always 0
-			if e.ObjectNew.GetGeneration() == 0 {
-				return true
-			}
-			return e.ObjectOld.GetGeneration() != e.ObjectNew.GetGeneration()
+			return e.ObjectOld.GetResourceVersion() != e.ObjectNew.GetResourceVersion()
```
Comment (clyang82) [inline operator/pkg/controllers/crd/crd_controller.go:275]: maybe detect generation + annotation changed. resourceversion can be too noisy. i remembered we met the problem that the resourceversion is updated continously

Comment (ldpliu) [reply operator/pkg/controllers/crd/crd_controller.go]: Done

**Review by clyang82 — APPROVED** [review]
LGTM

**Pull request merged**